src/error.rs

```diff
@@ -0,0 +1,24 @@
+use core::fmt;
+
+use thiserror::Error;
+
+/// Generic errors for WKT writing and reading
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Only 2D input is supported when writing Rect to WKT.")]
+    RectUnsupportedDimension,
+    #[error("Only defined dimensions and undefined dimensions of 2, 3, or 4 are supported.")]
+    UnknownDimension,
+    /// Wrapper around `[std::fmt::Error]`
+    #[error(transparent)]
+    FmtError(#[from] std::fmt::Error),
+}
+
+impl From<Error> for fmt::Error {
+    fn from(value: Error) -> Self {
+        match value {
+            Error::FmtError(err) => err,
+            _ => std::fmt::Error,
```

Comment:
nit: `std::fmt` here, but `core::fmt` at the top — settle on one.

Reply:
If we swallow the error details here, how does the user ever come to see the error details, e.g. "Only defined dimensions and undefined dimensions of 2, 3, or 4 are supported."

Reply:
Oh, oops. I don't actually know the difference between `std` and `core`